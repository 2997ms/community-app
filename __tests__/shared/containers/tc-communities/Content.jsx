import _ from 'lodash';
import React from 'react';
import Rnd from 'react-test-renderer/shallow';

const rnd = new Rnd();

const mockState = {
  tcCommunities: {
    meta: {
      communityId: 'someId',
      loading: false,
      leaderboardApiUrl: 'some/api/url',
    },
  },
};

const Content = require('containers/tc-communities/Content').default;

test('Matches shapshot', () => {
  rnd.render((
    <Content
      match={{
        params: {
          communityId: 'someId',
          pageId: 'somePageId',
          history: {},
          location: {},
        },
      }}
      store={{
        dispatch: () => _.noop,
        getState: () => mockState,
        subscribe: _.noop,
      }}
    />
  ));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
