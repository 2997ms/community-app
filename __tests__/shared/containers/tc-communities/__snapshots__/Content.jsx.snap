// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shapshot 1`] = `
<ContentContainer
  communityId="someId"
  dispatch={[Function]}
  isLoading={false}
  leaderboardApiUrl="some/api/url"
  loadedCommunityId="someId"
  match={
    Object {
      "params": Object {
        "communityId": "someId",
        "history": Object {},
        "location": Object {},
        "pageId": "somePageId",
      },
    }
  }
  pageId="somePageId"
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
